dashboard/config/libraries/Outbreak.interpreted.js

```diff
@@ -0,0 +1,158 @@
+    if (currentRecovery == 0) {
+      getHealthy(spriteIdArg);
+    } else {
+      setProp(spriteIdArg, "recovery", currentRecovery - 1);
+    }
+  }
+}
+
+/* --------------- END OF MONSTER BEHAVIOR DEFINITIONS --------------- */
+
+/* --------------- START OF LIBRARY LOGIC --------------- */
+// Owned By Curriculum
+var RECOVERY_TIME = 250;
+
+function getSickWithProbability(spriteIdArg, probability) {
+  if (randomNumber(0, 100) < probability) {
+    getSick(spriteIdArg);
+  }
+}
+
+checkTouching("when", {costume: "healthy"}, {costume: "sick"}, function (extraArgs) {
+  getSickWithProbability({id: extraArgs.subjectSprite}, 100);
+});
+
+checkTouching("when", {costume: "healthy_mask"}, {costume: "sick"}, function (extraArgs) {
+  getSickWithProbability({id: extraArgs.subjectSprite}, 75);
+});
+
+checkTouching("when", {costume: "healthy"}, {costume: "sick_mask"}, function (extraArgs) {
+  getSickWithProbability({id: extraArgs.subjectSprite}, 50);
+});
+
+checkTouching("when", {costume: "healthy_mask"}, {costume: "sick_mask"}, function (extraArgs) {
+  getSickWithProbability({id: extraArgs.subjectSprite}, 37.5);
+});
+/* ---------------- END OF LIBRARY LOGIC ---------------- */
+
+
+/* --------------- START OF BLOCKS API --------------- */
+// Owned By Star Labs
+
+function setupOutbreak(numMonsters, callback) {
+  var percentSick = 10;
+  var numSick = Math.round(percentSick * 0.01 * numMonsters);
+  makeNumSprites(numMonsters - numSick, "healthy");
+  makeNumSprites(numSick, "sick");
+  getSick({costume: "sick"}, "sick");
+  setProp({costume: "all"}, "scale", 25);
+  callback();
+}
+
+function layoutSprites(format) {
+  console.log(format);
+}
+
+function setSpeed(speed) {
+  setProp({costume: "all"}, "speed", speed);
+}
+
+function beginCongregating() {
+  addMonsterBehavior(congregatingBehavior);
+}
+
+function beginQuarantining() {
+  addMonsterBehavior(quarantiningBehavior);
+}
+
+function beginSocialDistancing() {
+  addMonsterBehavior(socialDistancingBehavior);
+}
+
+function beginWandering() {
+  addMonsterBehavior(wanderingBehavior);
+}
+
+function beginWearingMasks() {
+  setProp({costume: "all"}, "mask", true);
+  setAnimation({costume: "healthy"}, "healthy_mask");
+  setAnimation({costume: "sick"}, "sick_mask");
+}
+
+function stopMoving() {
+  removeBehaviorSimple({costume: "all"}, new Behavior(congregatingBehavior, []));
+  removeBehaviorSimple({costume: "all"}, new Behavior(socialDistancingBehavior, []));
+  removeBehaviorSimple({costume: "all"}, new Behavior(wanderingBehavior, []));
+}
+/* ---------------- END OF BLOCKS API ---------------- */
+
+/* --------------- START OF HELPER FUNCTIONS --------------- */
+// Owned By Star Labs
+function getSick(spriteIdArg) {
+  setProp(spriteIdArg, "sick", true);
+  var isWearingMask = getProp(spriteIdArg, "mask");
+  if (isWearingMask) {
+    setAnimation(spriteIdArg, "sick_mask");
+  } else {
+    setAnimation(spriteIdArg, "sick");
+  }
```

Comment:
stylistic nit: we could simplify these lines a little, like:

```javascript
var anim = getProp(spriteIdArg, "mask") ? "sick_mask" : "sick";
setAnimation(spriteIdArg, anim);
```

Reply:
(same below in `getHealthy`)

Reply:
Since these files will be jointly owned between Star Labs and Curriculum, I tried to keep the JS as readable as possible without using much special syntax. I'm not 100% sure what the right balance is, though, what do you think?

Reply:
ah, good consideration. i think it's okay to use ternaries here -- they're reasonably readable and this part of the file is owned by *labs